src/metabase/db/data_source.clj

```diff
@@ -10,6 +12,61 @@
   (:import java.sql.DriverManager
            java.util.Properties))
 
+(defn- h2? [url]
+  (.startsWith url "jdbc:h2:"))
+
+(defn- h2-path [url]
+  (let [[_ path] (re-matches #"jdbc:h2:file:(.*)$" url)]
+    path))
+
+(defn- h2-migration-paths [url]
+  (let [base (h2-path url)]
+    {:v1db   (str base ".mv.db")
+     :v2db   (str base ".v2.mv.db")
+     :script (str base ".v2migration.sql")}))
```

Comment:
We should probably make the script not a file in `$PWD` but in `/tmp` or something. What's the Right Way to do that that's eg. container-friendly?

Reply:
It would be really neat if we could do this entirely in-memory. Looks like you can run a script directly from a `java.io.Reader` but  http://h2database.com/javadoc/org/h2/tools/RunScript.html#execute-java.sql.Connection-java.io.Reader-- and maybe [`Script.process`](https://github.com/h2database/h2database/blob/master/h2/src/main/org/h2/tools/Script.java#L130-L137) could generate the script. Not sure if that's going to work well or not.

If not I think it's probably best to do it in the same directory as the H2 file itself -- after all we know we can write to that directory because that file is there